src/hazelcore/UndoHistoryCore.re

```diff
@@ -37,6 +37,7 @@ type action_group =
   /* SLine in Action_common.shape stands for both empty line and case rule,
      so an extra type CaseRule is added for construction */
   | CaseRule
+  | Subscript
```

Comment:
why do you need to special case subscript here?

Reply:
so that I can differentiate action_group when I construct `SLeftBracket` and show different keywords for subscript and list.